wvd-templates/DSC/Functions.ps1

```diff
@@ -444,28 +444,32 @@ function InstallRDAgents {
         throw "No registration token specified"
     }
 
-    RunMsiWithRetry -programDisplayName "RDAgentBootLoader" -isUninstall -argumentList @("/x {A38EE409-424D-4A0D-B5B6-5D66F20F62A5}", "/quiet", "/qn", "/norestart", "/passive") -msiOutputLogPath "C:\Users\AgentBootLoaderUnInstall.txt" -msiLogVerboseOutput:$EnableVerboseMsiLogging
-
-    while ($true) {
-        try {
-            $oldAgent = Get-Package -ProviderName msi -Name "Remote Desktop Services Infrastructure Agent"
-        }
-        catch {
-            #Ignore the error if it was due to no packages being found.
-            if ($PSItem.FullyQualifiedErrorId -eq "NoMatchFound,Microsoft.PowerShell.PackageManagement.Cmdlets.GetPackage") {
-                break
+    $msiNamesToUninstall = @(
+        @{ msiName = "Remote Desktop Services Infrastructure Agent"; displayName = "RD Infra Agent"; logPath = "C:\Users\AgentUninstall.txt"}, 
+        @{ msiName = "Remote Desktop Agent Boot Loader"; displayName = "RDAgentBootLoader"; logPath = "C:\Users\AgentBootLoaderUnInstall.txt"}
+    )

```

Comment:
Ref: https://github.com/Azure/RDS-Templates/blob/72c6ed157efe73588c9889384899afa136416662/wvd-templates/DSC/Functions.ps1#L367
Because we are using the same log for uninstallation, I think we need to add `/l+` argument to the list to append to the log file and preserve the output from previous uninstallation.  Dox: https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/msiexec#parameters-3

Reply:
I added that, thanks. On that same line I also added quotes around $msiOutputLogPath in case we use a path with spaces in the future.